src/librustc/mir/interpret/error.rs

```diff
@@ -228,6 +228,24 @@ impl<'tcx> From<InterpError<'tcx, u64>> for InterpErrorInfo<'tcx> {
 
 pub type AssertMessage<'tcx> = InterpError<'tcx, mir::Operand<'tcx>>;
 
+#[derive(Clone, RustcEncodable, RustcDecodable, HashStable)]
+pub enum EvalErrorPanic<O> {
```

Comment:
Name this enum `PanicMessage`. This way it makes more sense when we'll start reusing it in `TerminatorKind::Assert`'s `msg` field